### PR TITLE
feat: register upgrade for v4.0-rc3

### PR DESCRIPTION
**Pull request by adamewozniak**
## Description

Registers an upgrade from v4.0-rc1 to v4.0-rc3. This is a non-operating upgrade.

Necessary for a fix on canon-2 after the [rc3 pruning fix](https://github.com/umee-network/umee/pull/1720/files) is merged

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by gsk967** [comment]
@adamewozniak  why we need this upgrade handler because there is no migrations in the upgrade handler code

> do we need this for any specific reason 
> it is still in progress because there is no migrations in the upgrade handler code?

**Comment by gsk967** [comment]
@adamewozniak  https://github.com/umee-network/umee/pull/1720 in this pull request there is no change regarding the params or store , so i think we don't need upgrade handler for this, we can just swap the binaries

**Review by gsk967 — COMMENTED** [review]
@adamewozniak https://github.com/umee-network/umee/pull/1720 in this pull request there is no change regarding the params, store or any state change , so i think we don't need upgrade handler for this, we can just swap the binaries

**Comment by toteki** [comment]
> @adamewozniak #1720 in this pull request there is no change regarding the params, store or any state change , so i think we don't need upgrade handler for this, we can just swap the binaries

Binary swap would break (app hash mismatch) all validators who use canon-2 besides our test nodes. State breaking changes should use a coordinated upgrade

File: app/upgrades.go [hunk 1]
```diff
@@ -32,6 +32,18 @@ func (app UmeeApp) RegisterUpgradeHandlers(bool) {
 	app.registerUpgrade3_1to3_3(upgradeInfo)
 	app.registerUpgrade3_2to3_3(upgradeInfo)
 	app.registerUpgrade3_3to4_0(upgradeInfo)
+	app.registerUpgrade4_0to4_0rc3(upgradeInfo)
+}
+
+// performs upgrade from v4.0-rc2 -> v4.0-rc3
```
Comment (toteki) [inline app/upgrades.go:38]:
```suggestion
// performs upgrade from v4.0-rc2 (or rc1) -> v4.0-rc3
```

**Comment by adamewozniak** [comment]
> @adamewozniak #1720 in this pull request there is no change regarding the params, store or any state change , so i think we don't need upgrade handler for this, we can just swap the binaries

Yeah what @toteki said - even if there is no changed information, on coordinated upgrades you still need to define an upgrade plan

**Pull request merged**